vscode-smart-contracts/src/commands.ts

```diff
@@ -222,3 +237,36 @@ export async function initProject() {
   terminal.show();
   terminal.sendText(executable + " concordium init");
 }
+
+/** Regular expression for checking the output of `cargo --list` for the `generate` command. */
+const lineStartingWithGenerate = /^\s+generate[\s\n]/m;
+/**
+ * Check whether cargo-generate is available in PATH.
+ */
+async function haveCargoGenerateInstalled() {
+  const { stdout } = await execFile("cargo", ["--list"]);
+  console.log(stdout, lineStartingWithGenerate.test(stdout));
+  return lineStartingWithGenerate.test(stdout); // Check if the output have a line where the first word is `generate`.
+}
+
+/**
+ * Install the cargo-generate using cargo.
+ * The returned promise resolves when the install task has ended.
+ */
```

Comment:
Does this raise an exception if installation fails?

Reply:
How is that handled?

Reply:
It now returns the process exit code and fail if this is not 0.
I also ensured every failure is forwarded to the user, because VS Code will not show them by default.

Reply:
Great.